Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -296,21 +297,26 @@ protected void SetSyncVarGameObject(GameObject newGameObject, ref GameObject gam
                 }
             }
 
-            uint oldGameObjectNetId = 0;
-            if (gameObjectField != null)
+            // netId changed?
+            if (newGameObjectNetId != netIdField)
             {
-                oldGameObjectNetId = gameObjectField.GetComponent<NetworkIdentity>().netId;
-            }
-
-            if (newGameObjectNetId != oldGameObjectNetId)
-            {
-                if (LogFilter.Debug) { Debug.Log("SetSyncVar GameObject " + GetType().Name + " bit [" + dirtyBit + "] netfieldId:" + oldGameObjectNetId + "->" + newGameObjectNetId); }
+                if (LogFilter.Debug) { Debug.Log("SetSyncVar GameObject " + GetType().Name + " bit [" + dirtyBit + "] netfieldId:" + netIdField + "->" + newGameObjectNetId); }
                 SetDirtyBit(dirtyBit);
-                gameObjectField = newGameObject;
+                gameObjectField = newGameObject; // this is not really needed because we only access it via netId get/set anyway
```

Comment:
Ok,  I see we can still access the field directly in constructors and OnDeserialize.

This definitely needs a comment explaining it.

Reply:
it doesn't seem like we need this at all.
I just didn't remove it yet because NOT passing ref to setsyncvargameobject/netidentity BUT passing it to setsyncvar would make weaver code more complicated.